src/ngx_rewrite_driver_factory.cc

```diff
@@ -151,6 +151,10 @@ RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptions() {
   return options;
 }
 
+RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptionsForQuery() {
+  return new NgxRewriteOptions(thread_system());
```

Comment:
OK -- can you put in a note describing what you tried & discovered?  I am worried that other callers of NewRewriteOptions will have this bug as well given the place you did your fix.

Is there an equivalent in nginx to mod_instaweb.c's create_default_config ?